### ISSUE TITLE
Give User A Way to Re-build After Adding To packages.txt and/or requirements.txt without destroying data

### ISSUE THREAD

**Issue by andscoop** [astronomerio/astro-cli#74]
A user will need the ability to re-build their images after modifying `packages.txt` and/or `requirements.txt` without destroying volumes.

What do we want this user experience to be?

Some options to consider 
- `astro airflow refresh`
- `astro airflow start -r` to rebuild\refresh images
- try to persist volumes after a call to `astro airflow stop` so that next call to `astro airflow start` rebuilds images and keeps volumes. 

There seems to be a lot of ways to go about this - not all of them inline without docker works behind the scenes

For reference 
-------
I looked into [docker-compose down](https://docs.docker.com/compose/reference/down/) without passing the `-v` arg but this will result in an error `Volume "airflow_logs" needs to be recreated - driver has changed`. I believe this is because we are changing the underlying airflow container. In reality we only need the postgres container volumes to persist.

It looks like [docker-compose build](https://docs.docker.com/compose/reference/build/) could be leveraged to solve this issue.

**Comment by vparekh94** [comment]
As a user, I want to worry about packages/requirements as little as possible (spend my time focusing on using airflow itself). With that in mind, I think having `astro airflow stop` persist volumes and have the next call to `astro airflow start` rebuild and keep the volumes would be the best experience (assuming a new package or requirement was added, or something changed in the `Dockerfile`).

That being said, if this caused the image to be rebuilt every time (even if no new packages/requirements were added), I think having a `refresh` type command would be the best course of action since we've received feedback in the past from customers that our CLI can be slow. 

There might be value in having a `refresh` command anyways, as sometimes rebuilding docker images magically fixes everything.

*tl;dr:* From a user experience, both approaches are pretty solid, so I'd say whichever can be pushed out first :)

**Comment by schnie** [comment]
@andscoop this was the original idea behind the `astro airflow pause` command, which was just renamed to `astro airflow stop` right?

The intent was that stopping, would shutdown the containers, keeping the data volumes (postgres/logs) around. Starting back up would just mount those volumes and it'd look just as it was before stopping.

`astro airflow kill` was supposed to be the same, except killing the volumes (`-v`) this time.

Is this not how things are working now?

This was a bare bones approach that could be expanded upon though. I was thinking at one time that we could build some stuff to watch local files and trigger builds/restarts automatically, but I'm not sure how that would actually turn out in practice.

**Comment by andscoop** [comment]
@schnie Correct, `astro airflow pause` functionality is replaced by `astro airflow stop`

`astro airflow stop` will keep the volumes, but upon changing the underlying images and running libcompose equivalent `docker-compose up` we are getting errors similar to `Volume "postgres_data" needs to be recreated - driver has changed`.  I am going to continue digging into this, but am open to any ideas.

Error originating [here](https://github.com/astronomerio/astro-cli/blob/7ff80c696190433a2d9c12156465dce5ba222ae3/vendor/github.com/docker/libcompose/docker/volume/volume.go#L51) but what is not clear is why it is detecting a driver change when the drivers should remain the same.

**Comment by vparekh94** [comment]
@andscoop any update on this? It has been requested by a few customers so just want something to communicate back to them

**Comment by ryw** [comment]
@andscoop is this a task you can take on for 0.4 release?

**Comment by schnie** [comment]
@andscoop have you dug into this any more since we last chatted about it?

**Comment by andrewm4894** [comment]
howdy all, as a user, one issue i have here is having to install a new package or requirement and then really only being able to iterate on dev and test on a deployed cluster as opposed to my 'local' env which is much much slower.  

Also after a short while (about 3 new dags that needed package and lib changes) i end up with a local env that with errors like the below image. 

And then any time i need to do some dev that requires new packages or something it's like 10x slower process then developing on my 'local' server. 

Happy to help test or anything, just wanted to give an end user perspective. 

p.s. loving the fact this is all on git, really cool to have visibility on this stuff!

![image](https://user-images.githubusercontent.com/2178292/44117457-3783c77e-a00b-11e8-9dc3-3eeea242bad0.png)

**Comment by andscoop** [comment]
Update - 

Looks like there is a potential bug in `libcompose` where if no volume driver is set it does not assume the docker default of `local`. We can work around this by specifying the volume driver specifically. Will have a PR shortly

**Comment by andscoop** [comment]
This has been resolved and merged. Will be in next release.
